shell/apps/chrome-extension/extension-app-shell.js

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) 2017 Google Inc. All rights reserved.
+// This code may only be used under the BSD style license found at
+// http://polymer.github.io/LICENSE.txt
+// Code distributed by Google as part of this project is also
+// subject to an additional IP rights grant found at
+// http://polymer.github.io/PATENTS.txt
+
+import '../../app-shell/elements/arc-handle.js';
+import '../../apps/chrome-extension/chrome-data.js';
+
+import AppShell from '../../app-shell/app-shell.js';
+import ArcsUtils from '../../app-shell/lib/arcs-utils.js';
+import Xen from '../../components/xen/xen.js';
+
+class ExtensionAppShell extends AppShell {
```

Comment:
There may be a lot of stuff the base app shell does that we don't need for the extension. I totally support doing this for now and defer to @sjmiles review of this but maybe want to discuss with @sjmiles on potentially breaking apart or otherwise structuring so we only use what we need, ex. factoring classes differently or allowing composition vs inheritance, etc.

I could well be wrong, maybe the goal with the extension is for it to spin up a full blown app shell, but we might still think of an alternate way to hook in the extension specific tweaks.

Reply:
It's a good question. There are extension entry points (including this one) that do spin up a full interactive app shell, so this seems appropriate. However, we're discussing some other entry points that will just want suggestions, and I do wonder if we'll be able to optimize out some of the functionality. I expect we'll continue to discuss.